quickwit/quickwit-search/src/collector.rs

```diff
@@ -548,46 +550,27 @@ fn map_error(err: postcard::Error) -> TantivyError {
 }
 
 /// Merges a set of Leaf Results.
-fn merge_leaf_responses(
+fn merge_intermediate_aggregation_result<'a>(
     aggregations_opt: &Option<QuickwitAggregations>,
-    mut leaf_responses: Vec<LeafSearchResponse>,
-    sort_order1: SortOrder,
-    sort_order2: SortOrder,
-    max_hits: usize,
-) -> tantivy::Result<LeafSearchResponse> {
-    // Optimization: No merging needed if there is only one result.
-    if leaf_responses.len() == 1 {
-        return Ok(leaf_responses.pop().unwrap());
-    }
+    intermediate_aggregation_results: impl Iterator<Item = &'a [u8]>,
```

Comment:
```suggestion
    intermediate_aggregation_results: &dyn Iterator<Item = &'a [u8]>,
```

I think we don't need the generics here, do we?

Reply:
it's only monomorphized once for `FilterMap<std::slice::Iter, closure@>`, we can use dynamic dispatch, but I don't think there is an advantage to it in this case

Reply:
It gets two implementations. It is also called in `finalize`

Reply:
ho yes, mb